src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -154,4 +171,57 @@ private function getFile($id)
 
         return $this->directory.$hash[0].DIRECTORY_SEPARATOR.$hash[1].DIRECTORY_SEPARATOR.substr($hash, 2, -2);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doTag(array $tags)
+    {
+        $ok = true;
+
+        foreach ($tags as $tag => $ids) {
+            $file = $this->getFile($tag);
+            $dir = dirname($file);
+            if (!file_exists($dir)) {
+                @mkdir($dir, 0777, true);
+            }
+            $h = fopen($this->getFile($tag), 'ab');
+
+            foreach ($ids as $id) {
+                $ok = fwrite($h, rawurlencode($id)."\n") && $ok;
```

Comment:
I think it's better yes: easier to cde, and buffering is already handled by php internally.